Draft on data structure for transactions

Sh4pe opened the pull request:
This PR contains a proposal of the columns required to store the transaction raw data. It also contains some notes. The PR addresses some parts of the requirements for #5.

Please have a look and tell me what you think. It is very possible that I've missed something...

Sh4pe committed:
Add first draft of data structures proposal for transaction data

This draft contains a survey of columns provided by some possible
data sources.

Sh4pe committed:
Add columns for transaction raw data

fa-me commented on docs/data_structures/transactions.md:
```diff
@@ -24,4 +24,27 @@ data sources.
+
+**Notes**:
+
+* The unix timestamp `time` is universal since it can be used to quickly calculate differences and since it's timezone is well defined (i.e. UTC)
```
* the incoming data is probably only dates (relative to a timezone) but without explicit timezone - in order to convert it to unixtime we would have to assume or ask for a timezone. I would prefer if the software just goes with whatever timezone it is fed with. 
* the incoming data does not have a timestamp but only days - since the time is actually not really well defined in banking accounts (transactions take a while - you can rarely guarantee an hour at which the transaction will take place etc), we would have to make assumptions here
* it's not easily readable, making comparison between input data and database entry harder at an early stage of the project
* I don't see that calculating time differences is a very frequent or very expensive task
* if we need time differences, it will probably be more like "how many workdays are between X and Y", localized stuff, that will require 3rd party code anyways
* we throw away the original raw data, replacing it by newly formatted data while making assumptions about information that is actually not really important

Sh4pe replied:
You are right, I do agree with all the bullets you've mentioned. I've changed `time` to `date` in 2ba138c8122ce25e7e687dd75e0c8ef1208e336e.

fa-me commented on docs/data_structures/transactions.md:
```diff
@@ -24,4 +24,27 @@ data sources.
+**Notes**:
+
+* The unix timestamp `time` is universal since it can be used to quickly calculate differences and since it's timezone is well defined (i.e. UTC)
+* `sender_account` and `receiver_account` should contain everything required to uniquely identify the accounts involved. For transactions inside Europe, the IBAN should be sufficient.
```
"sender" and "sender_account" have a fixed relationship, so the data is redundant - we could think about having a seperate table where "sender"/"sender_account" relationships are listed

fa-me replied:
same holds for recipient of course

Sh4pe replied:
You are absolutely right. I've removed `sender` and `receiver` in 37a5a6ab8d2988caa89feaec02f1ebe85628c66d and added descriptions to `sender_account` and `receiver_account`. They should contain all information required to uniquely identify the sender/receiver.

fa-me commented on docs/data_structures/transactions.md:
```diff
@@ -24,4 +24,27 @@ data sources.
+| `text` | Text of the transaction |
+| `amount` | Amount of money that has been transferred |
+| `currency` | Currency of the transaction like "EUR" or "USD" |
+
```
we would need an additional column with "category" and/or "tags". Instead of listing the categories/tags, category/tag_id could be used

Sh4pe replied:
Oh - I've forgotten them. Added `category` and `tags` in 41813107311ddcfa5c58d2dda3a64ccf92788247.

I set the type of both `category` and `tags` to string, rather than choosing an ID as you proposed. The reason for this is that if we had IDs, we'd have to join another table/data source in order to get the name of the tags/categories. I think that the amount of data we'll be dealing with is small enough so that we do not need IDs here. I may be wrong, but I think that it'll be easier in the beginning at least.

And if it should become necessary to add IDs later, we can do so. What do you think?

fa-me replied:
yes, agreed

fa-me commented on docs/data_structures/transactions.md:
```diff
@@ -24,4 +24,27 @@ data sources.
-| MoneyMoney  | Banking application | Datum, Wertstellung, Kategorie, Name, Verwendungszweck, Konto, Bank, Betrag, Währung  |
\ No newline at end of file
+| MoneyMoney  | Banking application | Datum, Wertstellung, Kategorie, Name, Verwendungszweck, Konto, Bank, Betrag, Währung  |
+
```
thanks, very helpful to have this list

Sh4pe replied:
You're welcome :)

Sh4pe committed:
Change `time` to `date`

No timezone needs to be stored and calculating date differences
should be handled pretty well with Python code anyways.

Sh4pe committed:
Remove `sender` and `receiver`

The distinction between `sender` and `sender_account` was
redundant, since this could very well be achieved with only one
column. The same holds for `receiver`/`receiver_account` as well.

Update the description of `sender_account` and `receiver_account`
to include everything to uniquely identify the sender/receiver
as well.

Sh4pe committed:
Add `category` and `tags`
